src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -223,400 +452,399 @@ public void onUpgradeComplete(final int newLevel)
+    public void addPatrolTargets(final BlockPos target)
     {
-        return guardPos;
+        this.patrolTargets.add(target);
+        this.markDirty();
     }
 
     /**
-     * Set if the jobs should be assigned manually.
-     *
-     * @param assignManually the state to set.
+     * Resets the patrolTargets list.
      */
-    public void setAssignManually(final boolean assignManually)
+    public void resetPatrolTargets()
     {
-        this.assignManually = assignManually;
+        this.patrolTargets = new ArrayList<>();
         this.markDirty();
     }
 
     /**
-     * Possible job/AI.
+     * Getter for the bonus vision.
+     *
+     * @return an integer for the additional range.
      */
-    public enum GuardJob
+    public int getBonusVision()
     {
-        KNIGHT,
-        RANGER,
+        if (getBuildingLevel() <= MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return getBuildingLevel() * VISION_BONUS;
+        }
+        return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
     }
 
     /**
-     * Possible tasks.
+     * Populates the mobs list from the ForgeRegistries.
+     *
+     * @return the list of MobEntrys to attack.
      */
-    public enum Task
+    public List<MobEntryView> calculateMobs()
     {
-        FOLLOW,
-        GUARD,
-        PATROL
+        final List<MobEntryView> mobs = new ArrayList<>();
+
+        int i = 0;
+        for (final EntityEntry entry : ForgeRegistries.ENTITIES.getValuesCollection())
+        {
+            if (EntityMob.class.isAssignableFrom(entry.getEntityClass()))
+            {
+                i++;
+                mobs.add(new MobEntryView(entry.getRegistryName(), true, i));
+            }
+            else
+            {
+                for (final String location : Configurations.gameplay.guardResourceLocations)
+                {
+                    if (entry.getRegistryName() != null && entry.getRegistryName().toString().equals(location))
+                    {
+                        i++;
+                        mobs.add(new MobEntryView(entry.getRegistryName(), true, i));
+                    }
+                }
+            }
+        }
+
+        getColony().getPackageManager().getSubscribers().forEach(player -> MineColonies
+                                                                             .getNetwork()
+                                                                             .sendTo(new GuardMobAttackListMessage(getColony().getID(),
+                                                                               getID(),
+                                                                               mobsToAttack),
+                                                                               player));
+
+        return mobs;
     }
 
     /**
-     * The client view for the baker building.
+     * The client view for the Guard building.
      */
     public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 127 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)